### PR TITLE
Simplify policy lists header and its styles

### PR DESCRIPTION
**Pull request by josemigallas**
**What this PR does / why we need it**:

Policy lists are the chain and the registry. Form also shares the same header.
* Extracts a `HeaderButton` component
* Simplifie styles, removing some classes and combining repeated things.

**Verification steps**:
* `npm run update-dependencies`
* Navigate to `apiconfig/services/<id>/policies/edit`
* At Policy Chain button should be 'add':
Screen Shot 2019-11-18 at 12 54 12
* At Edit Policy or Select a Policy shuold be "cancel":
Screen Shot 2019-11-18 at 12 54 18

**Note**:
This is part 7 of the refactorization of Policies:
[THREESCALE-2221: Policies: Normalise the use of policies schema and its types](https://issues.jboss.org/browse/THREESCALE-2221)

Previous step #1408.

### CODE REVIEW

File: spec/javascripts/Policies/components/HeaderButton.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// @flow
+
+import React from 'react'
+import { mount } from 'enzyme'
+
+import { HeaderButton } from 'Policies/components/HeaderButton'
+
+it('should render itself', () => {
+  const wrapper = mount(<HeaderButton type='add' onClick={jest.fn()} />)
+  expect(wrapper.find(HeaderButton).exists()).toBe(true)
```
Comment (damianpm) [inline spec/javascripts/Policies/components/HeaderButton.spec.jsx]: I think we were starting to use jest snapshots

File: spec/javascripts/Policies/components/HeaderButton.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+it('should handle clicks', () => {
+  const spy = jest.fn()
+  const wrapper = mount(<HeaderButton type='add' onClick={spy} />)
+
+  wrapper.props().onClick()
+  expect(spy).toHaveBeenCalled()
+})
+
+it('should have different stules for each type', () => {
```
Comment (damianpm) [inline spec/javascripts/Policies/components/HeaderButton.spec.jsx]: Typo: "styles"
Or maybe better to rephrase like: "should render different icons"

File: spec/javascripts/Policies/components/HeaderButton.spec.jsx (hunk 1 above)
Comment (didierofrivia) [reply spec/javascripts/Policies/components/HeaderButton.spec.jsx]: Only when checking html structure, not just rendering

File: app/javascript/src/Policies/components/HeaderButton.jsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import type { ThunkAction } from 'Policies/types'
+
+type Props = {
+  type: 'add' | 'cancel',
+  onClick: () => ThunkAction,
+  children?: React.Node
+}
+
+const getIcon = type => {
+  switch (type) {
```
Comment (didierofrivia) [inline app/javascript/src/Policies/components/HeaderButton.jsx]: is there anything in lodash like Ramda's cond? https://ramdajs.com/docs/#cond ?

Comment (josemigallas) [reply app/javascript/src/Policies/components/HeaderButton.jsx]: Yes there is, the same function in fact.
https://www.npmjs.com/package/lodash.cond

Comment (josemigallas) [reply app/javascript/src/Policies/components/HeaderButton.jsx]: It doesn't look very good IMO:
```javascript
const getIcon = type => [
  [ type === 'add', 'fa fa-plus-circle' ],
  [ type === 'cancel', 'fa fa-times-circle' ]
]
```
I changed for a dictionary, I think it's looks simpler now.

File: spec/javascripts/Policies/components/HeaderButton.spec.jsx (hunk 1 above)
Comment (josemigallas) [reply spec/javascripts/Policies/components/HeaderButton.spec.jsx]: But you're right @damianpm, I'll add some snapshots

File: app/javascript/src/Policies/components/HeaderButton.jsx (hunk 3 above)
Comment (didierofrivia) [reply app/javascript/src/Policies/components/HeaderButton.jsx]: yeah, much better! 🏅

**Pull request merged**